stl/src/vector_algorithms.cpp

```diff
@@ -2382,37 +2944,39 @@ namespace {
 
                         if constexpr ((_Mode & _Mode_min) != 0) {
                             // Vector populated by the smallest element
-                            const auto _H_min     = _Traits::_H_min(_Cur_vals_min);
+                            const auto _H_min     = _H_min_wrap(_Cur_vals_min);
                             const auto _H_min_val = _Traits::_Get_any(_H_min); // Get any element of it
 
                             if (_H_min_val < _Cur_min_val) { // Current horizontal min is less than the old
                                 _Cur_min_val = _H_min_val; // update min
                                 // Mask of all elems eq to min
                                 const auto _Eq_mask = _Traits::_Cmp_eq(_H_min, _Cur_vals_min);
-                                unsigned long _Mask = _Traits::_Mask(_Traits::_Mask_cast(_Eq_mask));
+                                auto _Mask          = _Traits::_Mask(_Traits::_Mask_cast(_Eq_mask));
                                 // Indices of minimum elements or the greatest index if none
                                 const auto _Idx_min_val =
                                     _Traits::_Blend(_Traits::_All_ones(), _Cur_idx_min, _Traits::_Mask_cast(_Eq_mask));
                                 auto _Idx_min = _Traits::_H_min_u(_Idx_min_val); // The smallest indices
                                 // Select the smallest vertical indices from the smallest element mask
                                 _Mask &= _Traits::_Mask(_Traits::_Cmp_eq_idx(_Idx_min, _Idx_min_val));
-                                unsigned long _H_pos;
 
                                 // Find the smallest horizontal index
-
-                                // CodeQL [SM02313] _H_pos is always initialized: element exists, so _Mask != 0.
-                                _BitScanForward(&_H_pos, _Mask);
+                                unsigned long _H_pos = _Traits::_Get_first_h_pos(_Mask);
 
                                 // Extract its vertical index
+
+#ifdef _M_ARM64
+                                const auto _V_pos = _Traits::_Get_v_pos(_Idx_min);
+#else
                                 const auto _V_pos = _Traits::_Get_v_pos(_Cur_idx_min, _H_pos);
+#endif

```

Comment:
I think we can simplify and always use `_Idx_min`.

Not sure if we need to do this here or as a follow up.

Reply:
Happy to change here if we think it makes more sense than doing it separately?

Reply:
I think let's do this here.